CONTRIBUTING.md

```diff
@@ -0,0 +1,38 @@
+# Contributing to `ACCESS-OM2`s `spack` Environment
+
+## Via Pull Request
+
+### PRs for New Features
+
+For new features, the workflow is fairly straightforward.
+
+Simply open a pull request from your own branch to `main`, in which you make your modifications to the `spack.yaml` (and optionally the `config/versions.json`). The CI/CD will deploy these changes as a `Prerelease` version of `ACCESS-OM2` on `vk83/prerelease`, so you can verify that the changes will work before being merged formally into `main`.
+
+Some things to be mindful of:
+
+* Remember to update the version of the `access-om2` spec in the `spack.yaml` - this is the bit that looks like `- access-om2@git.VERSION`. It is this `VERSION` that will be eventually tagged to the commit.
+* The `config/versions.json` file is used to bundle the deployment with a particular version of `spack-packages` and `spack-config`. Don't forget to change this if there are cool new features in either of the repositories that you want to incorporate into your deployment.
+
+### PRs for Backported Bugfixes
+
+Since we use the `main` branch as a place for the bleeding-edge `ACCESS-OM2` changes, how do we go about backporting bugfixes?
+
+We use dedicated `backport/*.*` branches for bugfixes and additions to past versions of `ACCESS-OM2`. For example, say we have `2024.01.1` version of `ACCESS-OM2` on `main`, which we need to backport a bugfix.
+
+We should:
+
+* Branch off that commit with a `backport/2024.01` branch (if it doesn't already exist)
+* Open a PR off the `backport/2024.01` branch with the fixes, and when it is merged, will be tagged with `2024.01.2` on the `backport/2024.01` branch.
+
+We need to be mindful of the same things as merges of features into `main`:
+
+* Remember to update the version of the `access-om2` spec in the `spack.yaml` - this is the bit that looks like `- access-om2@git.VERSION`. It is this `VERSION` that will be eventually tagged to the commit.
+* The `config/versions.json` file is used to bundle the deployment with a particular version of `spack-packages` and `spack-config`. Don't forget to change this if there are cool new features in either of the repositories that you want to incorporate into your deployment.
```

Comment:
Kinda morbidly curious if this is possible, or something to be encouraged. Sounds deliciously dangerous to change the entire spack infrastructure for a bug fix, but then that might be what is required for said bug fix I guess.

I don't think we should disallow it, but maybe put some warning language around this being a pretty seat of the pants manoeuvre that should only be done if required to fix the bug.

Reply:
Added a note to this section

Reply:
No, it's here: https://github.com/ACCESS-NRI/ACCESS-OM2/pull/51/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R32-R33